### PR TITLE
Refactored prereqs makefiles to use apt repos.

### PR DESCRIPTION
**Pull request by johnkord**
Not sure if we want two different targets anymore ("make" and then "make install", according to the docs).  We could probably combine these targets now, at least for SGX1+FLC configurations

### CODE REVIEW

**Comment by johnkord** [comment]
bors try

**Comment by johnkord** [comment]
bors try

**Comment by johnkord** [comment]
bors try

File: prereqs/az-dcap-client/Makefile [hunk 1]
```diff
@@ -1,38 +1,18 @@
...
-
-build:
+setuprepo:
+	echo "deb [arch=amd64] https://packages.microsoft.com/ubuntu/16.04/prod xenial main" | tee /etc/apt/sources.list.d/msprod.list
+	wget -qO - https://packages.microsoft.com/keys/microsoft.asc | apt-key add -
+	apt-get update
 
 clean:
 
 install:
```
Comment (andschwa) [inline prereqs/az-dcap-client/Makefile]: Won't this fail if not run as `make && make install`? Since install doesn't depend on setuprepo.

Comment (andschwa) [reply prereqs/az-dcap-client/Makefile]: (Same for below. Also I might be wrong about make.)

Comment (johnkord) [reply prereqs/az-dcap-client/Makefile]: Right now the docs say to run:

```bash
sudo make -C prereqs/az-dcap-client
sudo make -C prereqs/az-dcap-client install

sudo make -C prereqs USE_LIBSGX=1
sudo make -C prereqs install USE_LIBSGX=1
```

I agree with you, I don't see a real need to put this in a makefile. We could make install-prereqs scripts that are specific for SGX1+FLC and SGX1

Comment (andschwa) [reply prereqs/az-dcap-client/Makefile]: I do think that GNU Make is fine for scripts like this (it's a nice scripting language when compared to Bash 😝 but make && make install is totally unnecessary. We should just add `setuprepo` as a dependency of install (and it should have `/etc/apt/sources.list.d/msprod.list` as a dependency so it only runs once).

**Pull request merged**